packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart

```diff
@@ -276,6 +277,91 @@ class CpuProfileData {
 
   factory CpuProfileData.empty() => CpuProfileData.parse({});
 
+  /// Generates [CpuProfileData] from the provided [CpuSamples].
+  ///
+  /// Only samples collected in the time range [origin, origin + extent] will be reported.
+  ///
+  /// [isolateId] The isolate id which was used to get the [cpuSamples].
+  /// This will be used to tag the stack frames and trace events.
+  /// [cpuSamples] The CPU samples that will be used to generate the [CpuProfileData]
+  static Future<CpuProfileData> generateFromCpuSamples(
```

Comment:
AFAICT, there's no asynchronous work being done in this method so this can be refactored into a factory. Maybe something like:

```
CpuProfileData.fromCpuSamples({required String isolateId, required vm_service.CpuSamples cpuSamples}) {...}
```

Reply:
oh true! this couldn't be a factory before because it had the async call to the vm_service but it has since been moved out. Great call!